app/models/piece.rb

```diff
@@ -16,4 +16,84 @@ class Piece < ActiveRecord::Base
   def types
     %w(Pawn Rook Knight Bishop Queen King)
   end
+
+  def obstructed?(destination_x, destination_y)
+    # initialize placeholder board
+    @board =
+      [
+        [0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0],
+        [0, 0, 1, 0, 0],
+        [0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0]
+      ]
+
+    # set common variables
+    @current_x = x_position
+    @current_y = y_position
+
+    @destination_x = destination_x
+    @destination_y = destination_y
+
+    @x_difference = @current_x - destination_x
+    @y_difference = @current_y - destination_y
```

Comment:
In models I recommend not using instance variables.

Instead, always pass the data you need as params to methods.

**Reasoning**: Code where the `state` of the data is contained in instance variables can become hard to follow. Imagine if one method updates an instance variable, then another method uses that variable for something. The results of your methods suddenly depend on the order it was called. Makes it hard to know where things change.

Better to have methods that take in parameters, because then you always know that "x input results in y output". 

The way this code is now "x input (+ the value of some instance variables), results in y output"

Make sense? :smile: